phase1-coordinator/src/tests.rs

```diff
@@ -1033,6 +1035,72 @@ fn coordinator_drop_contributor_clear_locks_test() -> anyhow::Result<()> {
     Ok(())
 }
 
+/// Drops a contributor and removes all subsequent contributions.
+#[test]
+#[serial]
+fn coordinator_drop_contributor_removes_subsequent_contributions() -> anyhow::Result<()> {
+    let parameters = Parameters::Custom((
+        ContributionMode::Chunked,
+        ProvingSystem::Groth16,
+        CurveKind::Bls12_377,
+        1, /* power */
+        2, /* batch_size */
+        2, /* chunk_size */
+    ));
+    let environment = initialize_test_environment_with_debug(&Testing::from(parameters).into());
+    let number_of_chunks = environment.number_of_chunks() as usize;
+
+    // Instantiate a coordinator.
+    let coordinator = Coordinator::new(environment, Box::new(Dummy))?;
+
+    // Initialize the ceremony to round 0.
+    coordinator.initialize()?;
+    assert_eq!(0, coordinator.current_round_height()?);
+
+    // Add a contributor and verifier to the queue.
+    let (contributor1, contributor_signing_key1, seed1) = create_contributor("1");
+    let (contributor2, contributor_signing_key2, seed2) = create_contributor("2");
+    let (verifier, verifier_signing_key) = create_verifier("1");
+    coordinator.add_to_queue(contributor1.clone(), 10)?;
+    coordinator.add_to_queue(contributor2.clone(), 9)?;
+    coordinator.add_to_queue(verifier.clone(), 10)?;
+
+    // Update the ceremony to round 1.
+    coordinator.update()?;
+
+    for _ in 0..number_of_chunks {
+        coordinator.contribute(&contributor1, &contributor_signing_key1, &seed1)?;
+        coordinator.contribute(&contributor2, &contributor_signing_key2, &seed2)?;
+        coordinator.verify(&verifier, &verifier_signing_key)?;
+        coordinator.verify(&verifier, &verifier_signing_key)?;
+    }
+
+    for (contributor, contributor_info) in coordinator.current_contributors() {
+        let expected_tasks = if contributor == contributor1 {
+            make_tasks(&[(0, 1), (1, 2)])
+        } else {
+            make_tasks(&[(1, 1), (0, 2)])
+        };
+        assert_eq!(contributor_info.assigned_tasks().len(), 0);
+        assert_eq!(contributor_info.completed_tasks(), &expected_tasks);
+    }
+
+    let locators = coordinator.drop_participant(&contributor1)?;
+    assert_eq!(2, locators.len());
+
+    for (contributor, contributor_info) in coordinator.current_contributors() {
+        if contributor == contributor2 {
```

Comment:
doing a `match` here might be a bit more explicit for what you're asserting

```rust
match contributor {
	contributor1 => {
		assert_eq!(contributor_info.completed_tasks().len(), 0);
    	        assert_eq!(contributor_info.assigned_tasks(), &make_tasks(&[(0, 1), (1, 2)]));
	}
	contributor2 => {
		assert_eq!(contributor_info.completed_tasks(), &make_tasks(&[(1, 1)]));
	        assert_eq!(contributor_info.assigned_tasks(), &make_tasks(&[(0, 2)]));
	}
	unexpected  => {
		panic!("current_contributors() contains an unexpected contributor: {:?}", unexpected)
	}
}
```

Reply:
That's very interesting, didn't know it was possible! I've tried it in playground and it doesn't work for some reason, can you check it?
https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=e7c3af9c3e1231c28364dc89067e653d

Reply:
@ibaryshnikov oooh okay, yeah you're right, I guess numbers/patterns need to be const in the match! Not sure how my brain mixed that one up. It looks like there [might still be a way to do it](https://stackoverflow.com/questions/55804275/are-there-traditional-style-switch-statements-in-rust).  https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=f9ee703a8cede19e064b4228b08bbeeb

```rust
match contributor {
	c if c == contributor1 => {
		assert_eq!(contributor_info.completed_tasks().len(), 0);
    	        assert_eq!(contributor_info.assigned_tasks(), &make_tasks(&[(0, 1), (1, 2)]));
	}
	c if c == contributor2 => {
		assert_eq!(contributor_info.completed_tasks(), &make_tasks(&[(1, 1)]));
	        assert_eq!(contributor_info.assigned_tasks(), &make_tasks(&[(0, 2)]));
	}
	unexpected  => {
		panic!("current_contributors() contains an unexpected contributor: {:?}", unexpected)
	}
}
```

Reply:
I just realised that the `else` block might not be matching `contributor1` but may also be matching the coordinator contributor, which seems to be a thing...

Reply:
Your comments helped me to improve the documentation and add a check for unexpected contributor, thanks!